lib/Stampie/Mailer.php

```diff
@@ -24,19 +29,24 @@
     protected $serverToken;
 
     /**
-     * @param AdapterInterface $adapter
-     * @param string           $serverToken
+     * @var MessageFactory
      */
-    public function __construct(AdapterInterface $adapter, $serverToken)
+    private $messageFactory;
+
+    /**
+     * @param HttpClient $adapter
+     * @param string     $serverToken
+     */
+    public function __construct(HttpClient $adapter = null, $serverToken)
```

Comment:
Please don't make this optional. let's force the user to inject what they need, and if they wish they can use the HttpClientDiscovery functionality on their own.